CMakeLists.txt

```diff
@@ -882,7 +882,7 @@ foreach(zephyr_lib ${ZEPHYR_LIBS_PROPERTY})
        AND NOT ${lib_imported}
     )
       get_property(allow_empty TARGET ${zephyr_lib} PROPERTY ALLOW_EMPTY)
-      if(NOT "${allow_empty}")
+      if(NOT ${allow_empty})
```

Comment:
Are you sure / what issue or false warning do you see ?
According to CMake docs: https://cmake.org/cmake/help/latest/command/if.html#string
> `if(<string>)`
>     A quoted string always evaluates to false unless:
>     - The string's value is one of the true constants, or

and the true constants are:
> True if the constant is 1, ON, YES, TRUE, Y, or a non-zero number (including floating point numbers)

So by keeping the quotes, then setting `ALLOW_EMPTY` to one of the above should result in the `if(NOT "y")` to become false and the warning not printed, but when .`ALLOW_EMPTY` is not set, a warning is printed.

However, the proposed `if(NOT ${allow_empty})` will expand into `if(NOT )` if `ALLOW_EMPTY` is not set which will evaluate to false, meaning default behavior will suddenly be to allow empty libraries.

If a library is empty and the warning should be silenced, then that library should set the `ALLOW_EMPTY` property to true.
Like this: https://github.com/zephyrproject-rtos/zephyr/blob/5876cc53d45201b0f7f185ae9f16c1e992422166/lib/posix/CMakeLists.txt#L48

Reply:
Thank you for your review. 
Somehow I mixed up my terminals and did not notice that my test did not produce the warning anymore where allow empty was not set. 

Nevertheless, I still have the warning for modules that are empty. But the issue for that is somewhere else...